network/tls_wrapper.go

```diff
@@ -19,15 +19,24 @@ package network
 import (
 	"crypto/tls"
 	"crypto/x509"
-	"io/ioutil"
-	"net"
-
 	"errors"
 	"github.com/cossacklabs/acra/logging"
 	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net"
 	"time"
 )
 
+// allowedCipherSuits that set in default tls config
+var allowedCipherSuits = []uint16{
+	tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
+	tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+	tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+}
```

Comment:
Is this list sorted by priority? If yes, then I would put it a bit differently:
* TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
* TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
* TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
* TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305

Justification:
- elliptic curves are more secure, in general faster than RSA and has smaller key sizes (hence the number of transferred bytes on the network is also lower), so they come before RSA
- AES comes before CHACHA - CHACHA is mostly useful to communicate with low-end mobile devices, which do not have AES crypto accelerators in their CPUs (non-hardware CHACHA is generally cheaper than AES and is mostly there for battery savings)
- AES-128 with SHA-256 come before AES-256 with SHA-384 (they have smaller key sizes and are usually better optimised, because they are more popular)

Reply:
agree, changed order in both cases (in second I copied order from first list and placed ecdsa before rsa in other cases)